### PR TITLE
[NestedTensor] Add jagged layout support for index_selectnestedtensor: add index_select support for jagged layout

### PR DESCRIPTION
**Pull request by xy200303**
## Summary

This PR adds `aten.index_select.default` support for jagged-layout NestedTensor (NJT).

Specifically, it:
- supports `index_select` on the batch dimension (`dim=0`)
- supports `index_select` on non-ragged dense dimensions
- keeps ragged-dimension `index_select` unsupported for now
- adds NJT OpInfo sample/reference coverage for `index_select`
- adds expected compile xfails for the current batch-dim implementation

## Motivation

`index_select` is a commonly used existing ATen operator, but jagged NestedTensor did not have dedicated support for it yet.

This change improves NestedTensor operator coverage while keeping the implementation aligned with the current NJT dispatch/testing patterns.

## Implementation details

- Registered `aten.index_select.default` in `torch/nested/_internal/ops.py`
- For `dim=0`, the implementation selects per-sequence segments and repacks them into a jagged NestedTensor
- For non-batch, non-ragged dims, the implementation forwards to `index_select` on the values buffer and preserves NJT metadata
- Added NJT-specific sample input generation and references in `torch/testing/_internal/opinfo/definitions/nested.py`
- Added expected compile coverage annotations for current batch-dim limitations

## Testing

Validated with:

- `python -m py_compile torch/nested/_internal/ops.py torch/testing/_internal/opinfo/definitions/nested.py test/test_nestedtensor.py`

I could not run the targeted NestedTensor test suite in this environment because a built/importable local `torch` was not available:
- `python test/test_nestedtensor.py -k index_select`

### CODE REVIEW

File: torch/testing/_internal/opinfo/definitions/nested.py [hunk 1]
```diff
@@ -864,6 +864,44 @@ def batchwise_reference_select(op, sample):
...
+        selected_starts = torch.index_select(njt.offsets()[:-1], 0, index)
+        selected_values = torch.cat(
+            [
+                torch.narrow(
+                    njt.values(),
+                    dim=njt._ragged_idx - 1,
+                    start=start.item(),
+                    length=length.item(),
+                )
+                for start, length in zip(selected_starts, selected_lengths)
```
Comment (Skylion007) [inline torch/testing/_internal/opinfo/definitions/nested.py]:
```suggestion
                for start, length in zip(selected_starts, selected_lengths, strict=True)

**Comment by zou3519** [comment]
Thank you very much for sending a PR.
To avoid you spending time preparing a PR for a feature we're not sure we want to add, our usual process is to discuss on the issue before doing the PR.
In particular, make sure that the issue is marked "actionable" before sending a PR!
See https://github.com/pytorch/pytorch/wiki/The-Ultimate-Guide-to-PyTorch-Contributions for details!
